docs/detections/rules-ui-create.asciidoc

```diff
@@ -263,35 +263,92 @@ For example, if a rule has an interval of 5 minutes, no additional look-back tim
 [[create-esql-rule]]
 === Create an ES|QL rule
 
-IMPORTANT: This is a placeholder for future documentation. The following content is incomplete.
+preview::[]
 
-. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
-. To create a rule that uses ES|QL, select **ES|QL**, 
-then write a query. There are two types of ES|QL queries:
+Use {ref}/esql.html[ES|QL] to query your source events and aggregate event data. Query results are returned in a table with rows and columns. Each row becomes an alert.
+
+To create an ES|QL rule:
 
-.. **Aggregating query**: This is a query that uses the `STATS...BY` grouping commands. Query results cannot be matched with a particular document in Elasticsearch. For example:
+. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
+. Select **ES|QL**, then write a <<esql-rule-query-types,query>>. 
 +
-[esql]
------
+TIP: Click the help icon (image:images/esql-ref-button.png[Click the ES|QL help icon,20,20]) to open the in-product reference documentation for all ES|QL commands and functions.
++
+. Click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.
+
+[float]
+[[esql-rule-query-types]]
+==== ES|QL query types 
+
+ES|QL rule queries are loosely categorized into two types: <<esql-agg-query,aggregating>> and <<esql-non-agg-query,non-aggregating>>. 
+
+[float]
+[[esql-agg-query]]
+===== Aggregating query 
+
+An aggregating query uses {ref}/esql-functions-operators.html#esql-agg-functions[`STATS...BY`] functions to aggregate source event data. Generated alerts only contain the fields returned by the query.
+
+Here is an example aggregating query:
+
+[source,esql]
+----
 FROM logs*
-| STATS count = COUNT(host.name) BY host.name
-| SORT host.name
+| STATS host_count = COUNT(host.name) BY host.name
+| SORT host_count DESC
+| WHERE host_count > 20
+----
+
+- This query starts by searching logs from indices that match the pattern `logs-*`. 
+- The query then aggregates the count of events by `host.name`.
+- Next, it sorts the result by `host_count` in descending order.
+- Then, it filters events where the `host_count` field is encountered more than 20 times during the specified rule interval.
+
+NOTE: Aggregating queries might create duplicate alerts if events that occurred in a rule's additional look-back time are aggregated during the rule's current and previous rule execution cycles.
```

Comment:
```suggestion
NOTE: Agrregating query rules may generate duplicate alerts, when events in additional look-back time getting aggregated twice: in current and previous rule executions
```

queries do not create alerts, alerts getting created during rule execution. ES|QL query just return results, from which alerts are generated

Reply:
Oh yeah, I can see how this could be interpreted as ES|QL queries generating alerts. I'm not sure that we want to use the term "agrregating query rules" though. It could be read as a new/different rule type. How about this:

```suggestion
NOTE: Rules that use aggregating queries might create duplicate alerts. This can happen  when events that occur in the additional look-back time are aggregated twice: once in the current rule execution and once in the a previous rule execution.
```

Reply:
@vitaliidm I'm merging this for now, but we can revisit it in a couple weeks if needs to adjusted for technical accuracy.

Reply:
agree, on aggr rquery rule terminlogy